### PR TITLE
feat: add playlist option

### PR DESCRIPTION
**Pull request by loiccoyle**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix
[ x] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**
I added the --playlist option which will open all the urls of the files in the torrent in a playlist in supported players. I could only get `mpv`, `mplayer` and `vlc` working. If you know how to do this for any of the other players I'm all ears.

**Which issue (if any) does this pull request address?**
#166 

**Is there anything you'd like reviewers to focus on?**
I didn't test the `vlc` on windows, its probably a good idea to do so. This option is most likely incompatible with the `--subtitles` option, I didn't enforce it though, maybe it should be checked and enforced?

I don't have much experience with JS so if I'm not following best practices, feel free to correct me!

### CODE REVIEW

**Comment by DiegoRBaquero** [comment]
Hi! Can you please re-base? I can review after it's clean

File: bin/cmd.js [hunk 1]
```diff
@@ -80,7 +81,7 @@ const playerArgs = {
   iina: ['/Applications/IINA.app/Contents/MacOS/iina-cli', '--keep-running'],
   mpv: ['mpv', '--really-quiet', '--loop=no'],
   mplayer: ['mplayer', '-really-quiet', '-noidx', '-loop', '0'],
-  smplayer: ['smplayer -close-at-end'],
+  smplayer: ['smplayer', '-close-at-end'],
```
Comment (loiccoyle) [inline bin/cmd.js:84]: Fixed a typo here !

**Comment by loiccoyle** [comment]
I merged upstream. I also fixed a typo in the `smplayer` args and I also added playlist support for `smplayer`.

File: bin/cmd.js [hunk 2]
```diff
@@ -52,6 +52,7 @@ const options = {
     q: { alias: 'quiet', desc: 'Don\'t show UI on stdout' },
     pip: { desc: 'Enter Picture-in-Picture if supported by the player' },
     verbose: { desc: 'Show torrent protocol details' },
+    'playlist': {desc: 'Open files in a playlist if supported by the player'},
```
Comment (DiegoRBaquero) [inline bin/cmd.js:55]:
```suggestion
    playlist: { desc: 'Open files in a playlist if supported by the player' },
```

File: bin/cmd.js [hunk 3]
```diff
@@ -404,18 +415,18 @@ function runDownload (torrentId) {
           return fatalError(err)
         }
         playerArgs.vlc[0] = vlcCmd
-        openPlayer(playerArgs.vlc.concat(JSON.stringify(href)))
+        argv.playlist ? openPlayer(playerArgs.vlc.concat(all_hrefs)) : openPlayer(playerArgs.vlc.concat(JSON.stringify(href)))
```
Comment (DiegoRBaquero) [inline bin/cmd.js:418]: don't they need to be JSON stringified?

Comment (loiccoyle) [reply bin/cmd.js:418]: Each href is `JSON.stringified` see line 397.

**Review by DiegoRBaquero — APPROVED** [review]

**Comment by loiccoyle** [comment]
Nice ! Thanks !

**Comment by eNV25** [comment]
I think this would work better if we generated an `.m3u` file from the URLs, instead of just passing URLs as arguments.

**Pull request merged**